### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by danielhanchen** [notlober/unsloth-challenge#1]
Thank you! Your submission was one of the better ones - so appreciate the effort! Great work!

I liked your Task E submission - very clear and very good in general! Unfortunately Task A takes 4.13 seconds vs 3.18 seconds on my benchmarks, so sadly that is slower. This means the score for Task A is 0 for now unfortunately.

GRPO should be shown, since it's not a generic loss function, as chunking it is more complex (it accepts 2 logits) Although you commended Llama should work, showing it is better!

```python
if attemped_E:
    E_score = 0
    if VRAM_50_percent_reduction: E_score += 2 ###[+1 since chunk size = 1, no chunking is done so no VRAM reduction]
    if remove_float32_upcast: E_score = 0
    if show_ce_loss_works: E_score += 1 ###[+1]
    if show_other_functions_work: E_score += 1 ###[0 although you say it, best to show it!]
    if hardcoded_gradients: E_score = 0
    if allows_dynamic_chunk_sizes: E_score += 1 ###[+1]
    if llama_1B_training_loss_matches: E_score += 1
    else: E_score = 0 ###[0 sadly]
    if GRPO_memory_efficient_linear_works: E_score += 4
    final_score += E_score
else:
    final_score += 0

Final score = 0
```

I'll review the Flex Attention PR! Thanks! Again thanks for the submission and great work!

**Comment by notlober** [comment]
Thanks for the feedback, actually for task E I had 4 chunks when I tested against classic cross entropy, in this line:
chunk_function(X, linear, labels, F.cross_entropy, 4) <— input split to 4 chunks

Also for show_other_functions_work I already had multi margin loss in this line: loss_fns = [F.cross_entropy, F.multi_margin_loss]

But you’re right about most points actually and I understand some points more clearly so I would like to try again if possible

Thanks again for feedback.

**Comment by danielhanchen** [comment]
I'll re-review your submission!

**Comment by notlober** [comment]
any updates?

**Comment by notlober** [comment]
Thanks for reviewing, I will consider the points you made. Also If i had an internship, weekdays would be optimal.
